src/services/BackgroundValidatePeers.ts

```diff
@@ -0,0 +1,38 @@
+import { Injector } from 'reduct'
+import PeerDatabase from './PeerDatabase'
+import axios from 'axios'
+import { create as createLogger } from '../common/log'
+import { shallowValidatePeer } from '../util/validatePeer'
+const log = createLogger('BackgroundValidatePeers')
+const CronJob = require('cron').CronJob
+
+export default class BackgroundValidatePeers {
+  private peerDb: PeerDatabase
+
+  constructor (deps: Injector) {
+    this.peerDb = deps(PeerDatabase)
+  }
+
+  start () {
+    const db = this.peerDb
+    const job = new CronJob({
+      cronTime: '00 30 2 * * *',
+      onTick: function () {
+        /*
+        * Runs every day at 2:30:00 AM.
+        */
+        log.debug('Validating Peers...')
+        const peers = db.getAllPeers()
+        peers.forEach(async (peer: string) => {
+          const peerInfo = await axios.get(peer + '/host-info')
```

Comment:
If one host fails to respond I believe axios will throw an error so this loop will stop, this should probably be in a try catch so one down host does not stop the full query.